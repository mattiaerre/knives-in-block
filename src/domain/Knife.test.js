import classNames from 'classnames';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { useState } from 'react';

function Knife({ color }) {
  const states = { block: 'block', down: 'down', up: 'up' };

  const [state, setState] = useState(states.down);

  function handleOnClick() {
    setState(states.up);
  }

  return (
    <div
      className={classNames(color, state)}
      data-testid="knife"
      onClick={handleOnClick}
    ></div>
  );
}

it('renders correctly', () => {
  const tree = create(<Knife color={'red'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('from down to up', async () => {
  render(<Knife color={'red'} />);
  await userEvent.click(screen.getByTestId('knife'));
  expect(screen.getByTestId('knife')).toHaveClass('up');
});
