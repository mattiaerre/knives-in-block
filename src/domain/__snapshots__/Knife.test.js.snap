// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="red down"
  data-testid="knife"
  onClick={[Function]}
/>
`;
